### ISSUE TITLE
Multi-line liquid syntax breaks when using < or > operators

### ISSUE THREAD

**Issue by cshold** [Shopify/liquid#1296]
This should be valid syntax, but causes the markup to be flagged as invalid (red opening `<div>` tag)
![image](https://user-images.githubusercontent.com/1730309/93550183-e1ccc680-f938-11ea-9f64-78d2ea913a53.png)

Same code with `=` instead of `<` gets proper formatting and no flag of broken HTML
![image](https://user-images.githubusercontent.com/1730309/93550241-fad57780-f938-11ea-89ea-8cd2c422cf82.png)

**Comment by cshold** [comment]
I originally thought this was just a highlighting error, but it also causes the 'broken HTML' notice in the Theme editor:
![image](https://user-images.githubusercontent.com/1730309/96607726-a66e4080-12c6-11eb-92c9-91d9d3c1b28b.png)


Recreate by adding this code to a product page:
```
{%- liquid
    assign inventory_visible = false
    if current_variant.inventory_management == 'shopify'
      if current_variant.inventory_quantity <= 10 and current_variant.inventory_quantity >= 0
        assign inventory_visible = true
      endif
    endif
    if current_variant.inventory_quantity == 0 or current_variant.inventory_policy == 'continue'
      assign inventory_visible = false
    endif
  -%}
```

You'll see the broken HTML notice. Swap with this and it does away:
```
{%- assign inventory_visible = false -%}
  {%- if current_variant.inventory_management == 'shopify' -%}
    {%- if current_variant.inventory_quantity <= 10 and current_variant.inventory_quantity >= 0 -%}
      {%- assign inventory_visible = true -%}
    {%- endif -%}
  {%- endif -%}
  {%- if current_variant.inventory_quantity == 0 or current_variant.inventory_policy == 'continue' -%}
    {%- assign inventory_visible = false -%}
  {%- endif -%}
```
```

**Comment by jonathanmoore** [comment]
For performance reasons, we're switching over to utilizing Liquid tags as much as possible.  Currently, this is one of the biggest holdups for broader adoption.

**Comment by dylanahsmith** [comment]
> I originally thought this was just a highlighting error, but it also causes the 'broken HTML' notice in the Theme editor:

Isn't that notice also a part of the theme editor?

Do you notice anything wrong with the HTML output itself?  If not, then this might be a theme editor bug and not a liquid bug.

**Comment by cshold** [comment]
In the code editor the syntax highlighting is the only issue.
In the theme editor it will give the "broken HTML" error.

Output of the HTML appears to be fine. No way for me to know where the bug originates, but it's a big pain and needs someone on the inside to div deeper. Code above should help.

**Comment by dylanahsmith** [comment]
Yes, I understand that the bug would be a pain.  I just wanted to make sure I understood the issue so it could be addressed in the right place.  I've brought it up internally and an issue has been opened for it.
